lib/extensions/ar_recent_records.rb

```diff
@@ -0,0 +1,26 @@
+module ArRecentRecords
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    # Returns a hash of counts by date for the :created_on field, with the
+    # date as the key and the count as the value.
+    #
+    # @param date Maximum number of days to look back. Default is 30.
+    # @param group_by Date group to use for collating data. Default is 'day'.
+    # @param filter Additional fields to add to the filter.
+    # @return [Hash] counts per group_by
+    #
+    # Examples:
+    #
+    #   Vm.recent_records                                       # => {'2018-10-01' => 5, '2018-10-02' => 3}
+    #   Vm.recent_records(10.days.ago.utc)                      # => {'2018-10-01' => 5, '2018-10-02' => 1}
+    #   Vm.recent_records(10.days.ago.utc, 'month')             # => {'2018-10-01' => 6}
+    #   Vm.recent_records(10.days.ago.utc, 'day', :ems_id => 1) # => {'2018-10-01' => 4, '2018-10-02' => 1}
+    #
+    def recent_records(date = 30.days.ago.utc, group_by = 'day', filter = {})
```

Comment:
One problem I have is that the name implies you are going to return those recent records, but instead you get counts.  Perhaps `recent_record_counts` instead?  Addtionally, adding a filter parameter feels wrong when ActiveRecord chaning should probably be used instead by the caller.  Here's an alternate layout that might solve those issues...what do you think?

```ruby
def recent_records(date = 30.days.ago.utc)
   where("created_on > ?", date)
end

def count_by_created_on(group_by = 'day')
   group("to_char(date_trunc('#{group_by}', created_on), 'YYYY-MM-DD')").count
end
```

Then the caller does...

```ruby
Vm.recent_records.where(filter).where(another_filter).count_by_created_on
```

Reply:
> Also I prefer the code returning an actual Time instance 😄 As a generic function I think it should do that. Can we remove the `to_char` stuff, or does that introduce too much overhead for what @NickLaMuro is trying to solve?

Well, to this comment, the result needed by the UI seems to be a string key that is in the format of `YYYY-MM-DD`, so I don't know if we can do it without the `to_char`. 

But maybe combining what you have in the comment above with a `format` option (defaulted to `nil`), and it will apply the `to_char` with the passed in format if it exists:

```ruby
Vm.recent_records(10.days.ago.utc).where(:ems_id => 1).count_by_created_on               # => {#<Time...> => 4, #<Time...> => 1}
Vm.recent_records(10.days.ago.utc).where(:ems_id => 1).count_by_created_on("YYYY-MM-DD") # => {'2018-10-01' => 4, '2018-10-02' => 1}
```

That work?

Reply:
@NickLaMuro Works for me. I'll implement both these suggestions.

Reply:
*minutes later*

Actually, I don't like these suggestions. I don't feel like doing it this way makes things any simpler than just doing it straight up with AR code.